.github/workflows/check-and-gen.yml

```diff
@@ -33,8 +33,6 @@ jobs:
       - uses: ./.github/actions/make-deps
       - run: make gen
```

Comment:
Probably too big a leap for now, but I'd actually like to move `docsgen` out of `make gen` so we can run it here. Maybe for now we just do docsgen-cli and if it works OK we'll consider expanding it.